### ISSUE TITLE
SmokeSensor names don't seem to be transferred from config.json to HomeKit

### ISSUE THREAD

**Issue by rrman** [jvmahon/homebridge-homeseer#92]
I just updated to the latest version of the homebridge plug-in. One issue I have observed is that the names of SmokeSensors don't seem to be transferred from config.json to HomeKit.
For example, I have SmokeSensor entries in my config.json file like:

        {
          "type": "SmokeSensor",
          "name": "Basement Smoke/CO",
          "ref": 101,
          "offValues": [
            255
          ]
        },
        {
          "type": "SmokeSensor",
          "name": "Kitchen Smoke/CO",
          "ref": 356,
          "offValues": [
            0
          ]
        },
...
... other SmokeSensors...
...

When a new bridge is added to HomeKit, the iOS Home App wizard shows each device that was added, and you can assign each device to the proper room. However, the Home App doesn't show the "name" that was assigned in config.json. For example, it does NOT show "Basement Smoke/CO" or "Kitchen Smoke/CO". The Home app just shows "Smoke Sensor". I have 5 Smoke Sensors so I couldn't tell which one was which.

Fortunately, after the iOS Home app finishes adding all the devices, I can go to each device's settings and I can check the "Serial Number" of the device, which says something like "HS SmokeSensor ref XXX" where XXX is the HomeSeer ref id. That's the way I can tell which room the Smoke Sensor belongs to.

To summarize: not a great big deal, but it seems that the name that I gave in config.json for each Smoke Sensor isn't transferred properly to HomeKit. All devices are just named "Smoke Sensor".

**Comment by jvmahon** [comment]
I'll take a look at this out over the weekend.

**Comment by rrman** [comment]
Note that I don't have the name issue for other types of devices. Only for SmokeSensor devices.

Regards.

**Comment by jvmahon** [comment]
Not sure what's up, but its working for me.  
Note that the "name" in config.json is only relevant the first time that HomeBridge starts with the referenced device. If you change the name in the iOS Home App,  the "new" name is what sticks, and the name in config.json is ignored. Similarly, if you first added the SmokeSensor without a name, such that the plugin gave it a default name, then you changed the name in config.json,  what you did in config.json will be ignored.

**Comment by rrman** [comment]
Okay, thanks for checking. The next time I have a need to recreate this particular bridge from scratch (probably not anytime soon, maybe when I move to new version of plug-in next year), I'll take careful notes to see if I can replicate this issue.  Thanks.
